streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -288,6 +288,22 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                 //  -> don't pass in `task.timeout.ms` to stay responsive if `KafkaStreams#close` gets called
                 final ConsumerRecords<byte[], byte[]> records = globalConsumer.poll(requestTimeoutPlusTaskTimeout);
                 if (records.isEmpty()) {
+                    // get consumer position to step over potential TX markers
+                    offset = retryUntilSuccessOrThrowOnTaskTimeout(
+                        () -> globalConsumer.position(topicPartition),
+                        String.format(
```

Comment:
I saw this `retryUntilSuccessOrThrowOnTaskTimeout` to get globalConsumer.position appear 3 times in this `restoreState` method. Could we extract them into a separate method?

Reply:
Good idea!